### ISSUE TITLE
Do not exec binary, use wkhtmltopdf library by cgo instead

### ISSUE THREAD

**Issue by szydell** [SebastiaanKlippert/go-wkhtmltopdf#19]
Wrapper for a cli tool is not the best way for writing such tool.
Please consider using [c library](https://wkhtmltopdf.org/libwkhtmltox/).
Static linking may result with a tool that can be delivered as a single binary without dependencies.

**Comment by SebastiaanKlippert** [comment]
To be honest, I have tried the C bindings first, but had some concurrency issues.
This was a few years back, so I don't remember the details but I think the issue was with libwkhtmltox itself. This may be solved now, I have not looked into this any further.
I see similar remarks on the web, from around the same time I looked into this:
* https://groups.google.com/forum/#!topic/wkhtmltopdf-general/CYm4kR08ia4/overview
* https://groups.google.com/forum/#!topic/golang-nuts/Unilbi-M-rY/overview
* https://github.com/umsatz/go-wkhtmltopdf

I have also not seen another library which uses libwkhtmltox bindings in Go, so I am not sure how well it works. If there is one, I would like to know.

But after doing some experients with this cmd wrapper we have found no issues and performance is good, so we decided to abandon the C bindings. 
When this wrapper was written we ran this in a worker environment and even with multiple concurrent calls we had no issues. But we currently run in AWS  Lambda, so there is only one process running at a time and I don't see any reason to improve this.

Also, at the time I wrote this, [AthenaPDF](https://www.athenapdf.com/) was not available, you might want to look at that if you don't like this solution. It is written in Go, so no bindings required.
